lib/segment/src/index/field_index/full_text_index/text_index.rs

```diff
@@ -47,7 +47,7 @@ impl FullTextIndex {
         })
     }
 
-    fn deserialize_document(data: &[u8], index: &mut InvertedIndex) -> OperationResult<Document> {
+    fn deserialize_document(data: &[u8]) -> OperationResult<BTreeSet<String>> {
```

Comment:
deserialization in my opinion has to have the same type as serialization. changed to type of serialization. Also `BTreeSet<_>` conversion avoids usage of `InvertedIndex` here and it makes the rusts borrow checker happier